libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -636,21 +638,28 @@ void AP_CRSF_Telem::process_device_info_frame(ParameterDeviceInfoFrame* info)
     // get the terminator of the device name string
     const uint8_t offset = strnlen((char*)info->payload,42U);
     if (strncmp((char*)info->payload, "Tracer", 6) == 0) {
-        _crsf_version.is_tracer = true;
+        _crsf_version.protocol = AP_RCProtocol_CRSF::ProtocolType::PROTOCOL_TRACER;
     } else if (strncmp((char*)&info->payload[offset+1], "ELRS", 4) == 0) {
         // ELRS magic number is ELRS encoded in the serial number
         // 0x45 'E' 0x4C 'L' 0x52 'R' 0x53 'S'
-        _crsf_version.is_elrs = true;
+        _crsf_version.protocol = AP_RCProtocol_CRSF::ProtocolType::PROTOCOL_ELRS;
+    }
+
+    if (!is_elrs()) {
+        /*
+            fw major ver = offset + terminator (8bits) + serial (32bits) + hw id (32bits) + 3rd byte of sw id = 11bytes
+            fw minor ver = offset + terminator (8bits) + serial (32bits) + hw id (32bits) + 4th byte of sw id = 12bytes
+        */
+        _crsf_version.major = info->payload[offset+11];
+        _crsf_version.minor = info->payload[offset+12];
+    } else {
+        // ELRS does not populate the version field so cook up something sensible
+        _crsf_version.major = 1;
```

Comment:
not sure this is correct, ELRS has versions as well, printing 1.0 would be misleading

Reply:
0 is also misleading - its only printed at debug, can probably just leave